### DISCUSSION TITLE
Supported architectures and ARCH variable?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AppImage/AppImageKit#1241] [Q&A]
I'd like to confirm that appimagetool detects the architecture of supplied binaries automatically?

I understand that the architecture may be supplied as ARCH environment variable (i.e. as "x86_64" etc). If correct, do we have a list of recognised values?

**Comment by TheAssassin** [comment]
It does. Every appimagetool ships with a single runtime only, though. You can't build an AppImage for `aarch64` with the `x86_64` AppImage. If you provide the right runtime (there's a parameter for appimagetool), you can build an `aarch64` AppImage on a `x86_64` machine using the `x86_64` AppImage.

Typically, nobody does that, though. For instance, our `aarch64` AppImages are built in a suitable environment from start to end. There is no reason to build the AppImage on a different architecture. Not even when using `qemu-user`'s `binfmt_misc` integration to seamlessly use QEMU to run binaries built for architectures the native CPU can't work with.
